File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/wiget/WidgetSettingsPresenter.java
package treehou.se.habit.ui.settings.subsettings.wiget;

import javax.inject.Inject;

import io.realm.Realm;
import treehou.se.habit.core.db.settings.WidgetSettingsDB;
import treehou.se.habit.module.RxPresenter;

public class WidgetSettingsPresenter extends RxPresenter implements WidgetSettingsContract.Presenter {

    private WidgetSettingsContract.View view;
    private Realm realm;

    @Inject
    public WidgetSettingsPresenter(WidgetSettingsContract.View view, Realm realm) {
        this.view = view;
        this.realm = realm;
    }

    @Override
    public void setWidgetBackground(int backgroundType) {
        WidgetSettingsDB settings = WidgetSettingsDB.loadGlobal(realm);
        realm.beginTransaction();
        settings.setImageBackground(backgroundType);
        realm.commitTransaction();
        view.setWidgetBackground(backgroundType);
    }

    @Override
    public void setWidgetTextSize(int size) {
        final WidgetSettingsDB settings = WidgetSettingsDB.loadGlobal(realm);
        realm.beginTransaction();
        settings.setTextSize(size);
        realm.commitTransaction();
        view.setWidgetTextSize(size);
    }

    @Override
    public void setWidgetImageSize(int size) {
        final WidgetSettingsDB settings = WidgetSettingsDB.loadGlobal(realm);
        realm.beginTransaction();
        settings.setIconSize(size);
        realm.commitTransaction();
        view.setWidgetImageSize(size);
    }

    @Override
    public void setCompressedWidgetSlider(boolean isChecked) {
        WidgetSettingsDB settings = WidgetSettingsDB.loadGlobal(realm);
        realm.beginTransaction();
        settings.setCompressedSlider(isChecked);
        realm.commitTransaction();
        view.setCompressedWidgetSlider(isChecked);
    }

    @Override
    public void setCompressedWidgetButton(boolean isChecked) {
        WidgetSettingsDB settings = WidgetSettingsDB.loadGlobal(realm);
        realm.beginTransaction();
        settings.setCompressedSingleButton(isChecked);
        realm.commitTransaction();
        view.setCompressedWidgetButton(isChecked);
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/wiget/WidgetSettingsPresenter.kt
package treehou.se.habit.ui.settings.subsettings.wiget

import javax.inject.Inject

import io.realm.Realm
import treehou.se.habit.core.db.settings.WidgetSettingsDB
import treehou.se.habit.dagger.RxPresenter

class WidgetSettingsPresenter @Inject
constructor(private val view: WidgetSettingsContract.View, private val realm: Realm) : RxPresenter(), WidgetSettingsContract.Presenter {

    override fun setWidgetBackground(backgroundType: Int) {
        val settings = WidgetSettingsDB.loadGlobal(realm)
        realm.beginTransaction()
        settings.imageBackground = backgroundType
        realm.commitTransaction()
        view.setWidgetBackground(backgroundType)
    }

    override fun setWidgetTextSize(size: Int) {
        val settings = WidgetSettingsDB.loadGlobal(realm)
        realm.beginTransaction()
        settings.textSize = size
        realm.commitTransaction()
        view.setWidgetTextSize(size)
    }

    override fun setWidgetImageSize(size: Int) {
        val settings = WidgetSettingsDB.loadGlobal(realm)
        realm.beginTransaction()
        settings.iconSize = size
        realm.commitTransaction()
        view.setWidgetImageSize(size)
    }

    override fun setCompressedWidgetSlider(isChecked: Boolean) {
        val settings = WidgetSettingsDB.loadGlobal(realm)
        realm.beginTransaction()
        settings.isCompressedSlider = isChecked
        realm.commitTransaction()
        view.setCompressedWidgetSlider(isChecked)
    }

    override fun setCompressedWidgetButton(isChecked: Boolean) {
        val settings = WidgetSettingsDB.loadGlobal(realm)
        realm.beginTransaction()
        settings.isCompressedSingleButton = isChecked
        realm.commitTransaction()
        view.setCompressedWidgetButton(isChecked)
    }
}
